lib/matplotlib/collections.py

```diff
@@ -1989,22 +2019,55 @@ class QuadMesh(Collection):
     vertex at mesh coordinates (0, 0), then the one at (0, 1), then at (0, 2)
     .. (0, meshWidth), (1, 0), (1, 1), and so on.
 
-    *shading* may be 'flat', or 'gouraud'
     """
-    def __init__(self, meshWidth, meshHeight, coordinates,
-                 antialiased=True, shading='flat', **kwargs):
+    def __init__(self, *args, **kwargs):
+        # signature deprecation since="3.5": Change to new signature after the
+        # deprecation has expired. Also remove setting __init__.__signature__,
+        # and remove the Notes from the docstring.
+        #
+        # We use lambdas to parse *args, **kwargs through the respective old
+        # and new signatures.
+        try:
+            # Old signature:
+            # The following raises a TypeError iif the args don't match.
```

Comment:
Maybe explain a little bit what this does?
```suggestion
            # The following raises a TypeError iif the args don't match.  The lambda tries to 
            # assign the *args, and **kwargs to the old-style signature and fails if the 
            # new style signature is being used.  
```

Reply:
We have the comment above
```
       # We use lambdas to parse *args, **kwargs through the respective old
        # and new signatures.
```
which I think is good enough given that this is just a temporary solution during the deprecatoin.

Reply:
Ooops sorry missed that...